From: SJ Park <sj@kernel.org>
Date: Thu, 27 Aug 2026 07:23:04 -0700
Subject: [PATCH] Revert "mm/damon/core: initialize
 damos_quota_goal->last_psi_total"

This reverts commit 440387fac4995c08eae9f9bc7ba44e83f9959583.

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 4ac21a10daf74..04af826704927 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1271,15 +1271,10 @@ static struct damos_quota_goal *damos_nth_quota_goal(
 	return NULL;
 }
 
-static u64 damos_get_some_mem_psi_total(void);
-
 static void damos_commit_quota_goal_union(
 		struct damos_quota_goal *dst, struct damos_quota_goal *src)
 {
 	switch (dst->metric) {
-	case DAMOS_QUOTA_SOME_MEM_PSI_US:
-		dst->last_psi_total = damos_get_some_mem_psi_total();
-		break;
 	case DAMOS_QUOTA_NODE_MEM_USED_BP:
 	case DAMOS_QUOTA_NODE_MEM_FREE_BP:
 		dst->nid = src->nid;
@@ -1307,6 +1302,7 @@ static int damos_commit_quota_goal(
 	dst->target_value = src->target_value;
 	if (dst->metric == DAMOS_QUOTA_USER_INPUT)
 		dst->current_value = src->current_value;
+	/* keep last_psi_total as is, since it will be updated in next cycle */
 	damos_commit_quota_goal_union(dst, src);
 	return 0;
 }
@@ -4111,17 +4107,6 @@ static int kdamond_wait_activation(struct damon_ctx *ctx)
 	return -EBUSY;
 }
 
-static void damos_init_quota_goal_last_psi(struct damos *s)
-{
-	struct damos_quota_goal *goal;
-
-	damos_for_each_quota_goal(goal, &s->quota) {
-		if (goal->metric != DAMOS_QUOTA_SOME_MEM_PSI_US)
-			continue;
-		goal->last_psi_total = damos_get_some_mem_psi_total();
-	}
-}
-
 static void kdamond_init_ctx(struct damon_ctx *ctx)
 {
 	unsigned long sample_interval = ctx->attrs.sample_interval ?
@@ -4139,7 +4124,6 @@ static void kdamond_init_ctx(struct damon_ctx *ctx)
 	damon_for_each_scheme(scheme, ctx) {
 		damos_set_next_apply_sis(scheme, ctx);
 		damos_set_filters_default_reject(scheme);
-		damos_init_quota_goal_last_psi(scheme);
 	}
 }
 
-- 
2.47.3

